src/libostree/ostree-repo-private.h

```diff
@@ -126,6 +129,24 @@ typedef struct {
   char checksum[OSTREE_SHA256_STRING_LEN+1];
 } OstreeDevIno;
 
+/* A MemoryCacheRef is an in-memory cache of objects (currently just DIRMETA).  This can
+ * be used when performing an operation that traverses a repository in someway.  Currently,
+ * the primary use case is ostree_repo_checkout_at() avoiding lots of duplicate dirmeta
+ * lookups.
+ */
+typedef struct {
+  OstreeRepo *repo;
+} OstreeRepoMemoryCacheRef;
```

Comment:
I'm confused why we need this, since the cache itself is part of `OstreeRepo`. It seems like the goal is to make the cache have a different lifetime from the `OstreeRepo` object? (Though it looks like the cache shouldn't be that large). Or was including it in `OstreeRepo` just a hack to avoid plumbing through a new set of arguments everywhere? Just making sure I understand the intent here.

I.e. can't we just always initialize the hash table and drop the secondary refcounting? This would have the advantage of reusing the cache across checkouts as well.